caffe2/python/trt/test_pt_onnx_trt.py

```diff
@@ -0,0 +1,258 @@
+    def normalize_image(image):
+        c, h, w = input_shape
+        image_arr = np.asarray(image.resize((w, h), Image.ANTIALIAS)).transpose([2, 0, 1])\
+            .astype(trt.nptype(trt.float32)).ravel()
+        if (normalization_hint == 0):
+            return (image_arr / 255.0 - 0.45) / 0.225
+        elif (normalization_hint == 1):
+            return (image_arr / 256.0 - 0.5)
+    np.copyto(pagelocked_buffer, normalize_image(Image.open(test_image)))
+    return test_image
+
+class Test_PT_ONNX_TRT(unittest.TestCase):
+    def __enter__(self):
+        return self
+
+    def setUp(self):
+        data_path = os.path.join(os.path.dirname(__file__), 'data')
+        self.image_files=["binoculars.jpeg", "reflex_camera.jpeg", "tabby_tiger_cat.jpg"]
+        for index, f in enumerate(self.image_files):
+            self.image_files[index] = os.path.abspath(os.path.join(data_path, f))
+            if not os.path.exists(self.image_files[index]):
+                raise FileNotFoundError(self.image_files[index] + " does not exist.")
+        self.labels = open(os.path.abspath(os.path.join(data_path, "class_labels.txt")), 'r').read().split('\n')
+
+    def build_engine_onnx(self, model_file):
+        with trt.Builder(TRT_LOGGER) as builder, builder.create_network(flags = 1) as network, trt.OnnxParser(network, TRT_LOGGER) as parser:
+            builder.max_workspace_size = 1 << 33
+            with open(model_file, 'rb') as model:
+                if not parser.parse(model.read()):
+                    for error in range(parser.num_errors):
+                        self.fail("ERROR: {}".format(parser.get_error(error)))
+            return builder.build_cuda_engine(network)
+
+    def _test_model(self, model_name, input_shape = (3, 224, 224), normalization_hint = 0):
+        if model_name == "resnet18":
+            model = models.resnet.resnet18(pretrained=True)
+        elif model_name == "resnet34":
+            model = models.resnet.resnet34(pretrained=True)
+        elif model_name == "resnet50":
+            model = models.resnet.resnet50(pretrained=True)
+        elif model_name == "resnet101":
+            model = models.resnet.resnet101(pretrained=True)
+        elif model_name == "resnet152":
+            model = models.resnet.resnet152(pretrained=True)
+
+        elif model_name == "resnet50_2":
+            model = models.resnet.wide_resnet50_2(pretrained=True)
+        elif model_name == "resnet101_2":
+            model = models.resnet.wide_resnet101_2(pretrained=True)
+
+        elif model_name == "googlenet":
+            model = models.googlenet(pretrained=True)
+
+        elif model_name == "squezenet1_0":
+            model = models.squeezenet1_0(pretrained=True)
+        elif model_name == "squezenet1_1":
+            model = models.squeezenet1_1(pretrained=True)
+
+        elif model_name == "inception_v3":
+            model = models.inception_v3(pretrained=True)
+
+        elif model_name == "mnasnet0_5":
```

Comment:
shall we create a map instead having so many if else branches?

Reply:
Fixed, thanks!